functions/Measure-DbaDiskSpaceRequirement.ps1

```diff
@@ -0,0 +1,355 @@
+
+            To connect as a different Windows user, run PowerShell as that user.
+
+        .PARAMETER SourceDatabase
+            The database to copy. It MUST exist.
+
+        .PARAMETER Destination
+            The destination SQL Server instance.
+
+        .PARAMETER DestinationSqlCredential
+            Same as SourceSqlCredential.
+
+        .PARAMETER DestinationDatabase
+            The database name at destination. May or may not be present. Unspecified name will assume database name be the same as source.
+
+        .PARAMETER Credential
+            Windows credentials to connect via CIM/WMI/PowerShell remoting for MountPoint definition.
+
+        .PARAMETER Consolidate
+            Will summarize space by ComputerName and MountPoints.
+
+        .NOTES
+            Tags: Migration
+
+            Website: https://dbatools.io
+            Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+            License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+        .LINK
+            https://dbatools.io/Measure-DbaDiskSpaceRequirement
+
+        .EXAMPLE
+            Measure-DbaDiskSpaceRequirement -SI INSTANCE1 -SD DB1 -DI INSTANCE2 -Consolidate
+
+            Calculate a simple migration with one database with then same name at destination. In this scenario,
+            no space is required since we will recover 448 KB on destination disk.
+
+            DatabaseName1   DatabaseName2   Name1     Name2   SizeKB1 SizeKB2 DiffKB ComputerName MountPoint
+            -------------   -------------   -----     -----   ------- ------- ------ ------------ ----------
+            DB1             DB1             DB1       DB1        3072   -3000     72 INSTANCE2    D:\
+            DB1             DB1             DB1_log   DB1_log     504   -1024   -520 INSTANCE2    D:\
+
+            ComputerName MountPoint RequiredSpaceKB
+            ------------ ---------- ---------------
+            INSTANCE2    D:\                   -448
+
+        .EXAMPLE
+            @([PSCustomObject]@{Source='SQL1';Destination='SQL2';Database='DB1'},
+              [PSCustomObject]@{Source='SQL1';Destination='SQL2';Database='DB2'}
+            ) | Measure-DbaDiskSpaceRequirement -Consolidate
+
+            Using a PSCustomObject with 2 databases to migrate
+
+        .EXAMPLE
+            Import-Csv -Path .\migration.csv -Delimiter "`t" | Measure-DbaDiskSpaceRequirement -Consolidate
+
+            Using a CSV file. You will need a header in migration.csv "Source<tab>Destination<tab>Database"
+
+        .EXAMPLE
+            Invoke-DbaSqlCmd -SqlInstance DBA -Database Migrations -Query 'select Source,Destination,DatabaseName from refresh.Migrations' `
+                | Measure-DbaDiskSpaceRequirement -Consolidate -Verbose
+
+            Using a SQL table. We are DBA after all!
+    #>
+
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('SI','SqlInstance')]
+        [string]$Source,
```

Comment:
This should be `DbaInstanceParameter`, and the aliases can be removed.

Reply:
You want me to remove alias but I find alias in 416 files out of 752 which is more than half. They provide interface with the pipeline. I can remove the shortest one because they are really meant to shorten the call. But SqlInstance is THE way to call an instance in dbatools.

Reply:
`SqlInstance` is used only when you are working with one server. This server utilizes parameters for source and destination, which the commands we have with those parameters do not include aliases.